File: scratch/fix_companions.dart
import 'dart:convert';
import 'dart:io';

void main() {
  final file = File(r'c:\Users\us mohamed\Desktop\مصحف\assets\data\companions.json');
  if (!file.existsSync()) {
    print('File not found!');
    return;
  }
  
  try {
    final content = file.readAsStringSync();
    final List<dynamic> data = json.decode(content);
    
    bool found = false;
    for (var item in data) {
      if (item['id'] == 19) {
        print('Fixing ID 19 name: ${item['name']} -> خالد بن الوليد');
        item['name'] = 'خالد بن الوليد';
        found = true;
      }
    }
    
    if (found) {
      final encoder = JsonEncoder.withIndent('  ');
      file.writeAsStringSync(encoder.convert(data));
      print('companions.json updated successfully!');
    } else {
      print('ID 19 not found in JSON!');
    }
  } catch (e) {
    print('Error: $e');
  }
}

File: scratch/fix_companions.py
import json

file_path = r"c:\Users\us mohamed\Desktop\مصحف\assets\data\companions.json"

try:
    with open(file_path, 'r', encoding='utf-8') as f:
        data = json.load(f)
        
    for companion in data:
        if companion.get("id") == 19:
            print(f"Fixing ID 19 name: {companion.get('name')} -> خالد بن الوليد")
            companion["name"] = "خالد بن الوليد"
            
    with open(file_path, 'w', encoding='utf-8') as f:
        json.dump(data, f, ensure_ascii=False, indent=2)
    print("companions.json fixed and formatted successfully.")
except Exception as e:
    print(f"Error: {e}")
